### PR TITLE
Made machine control covers disable themselves upon power loss

### PR DESCRIPTION
**Pull request by johnch18**
Previously an activated machine controller cover would continuously try to run a recipe even if the multiblock had run out of energy, burning through resources every tick.
This change makes a machine controller attached to a tile entity that has run out of power change its state to "Disable Machine" in order to prevent further resource burn.

### CODE REVIEW

File: src/main/java/gregtech/api/metatileentity/BaseMetaTileEntity.java [hunk 1]
```diff
@@ -92,6 +92,8 @@
     private UUID mOwnerUuid = GT_Utility.defaultUuid;
     private NBTTagCompound mRecipeStuff = new NBTTagCompound();
 
+    public boolean _wasShutdown = false;
```
Comment (botn365) [inline src/main/java/gregtech/api/metatileentity/BaseMetaTileEntity.java:95]: naming should be somhat consistent this should be `mWasShutdown`

**Review by botn365 — CHANGES_REQUESTED** [review]

File: src/main/java/gregtech/api/metatileentity/BaseMetaTileEntity.java (hunk 1 above)
Comment (johnch18) [reply src/main/java/gregtech/api/metatileentity/BaseMetaTileEntity.java]: Gotcha

**Review by botn365 — CHANGES_REQUESTED** [review]

File: src/main/java/gregtech/common/covers/GT_Cover_ControlsWork.java [hunk 2]
```diff
@@ -16,11 +16,16 @@
 public class GT_Cover_ControlsWork extends GT_CoverBehavior {
     public int doCoverThings(byte aSide, byte aInputRedstone, int aCoverID, int aCoverVariable, ICoverable aTileEntity, long aTimer) {
         if (aTileEntity instanceof IMachineProgress) {
-            if ((aInputRedstone > 0) == (aCoverVariable == 0) && aCoverVariable != 2)
-                ((IMachineProgress) aTileEntity).enableWorking();
-            else
+            if (!((IMachineProgress) aTileEntity).wasShutdown()) {
```
[the remark this answers is not in the document]
Comment (johnch18) [reply src/main/java/gregtech/common/covers/GT_Cover_ControlsWork.java:19]: Like in the configs or as a parameter?

[the remark this answers is not in the document]
Comment (botn365) [reply src/main/java/gregtech/common/covers/GT_Cover_ControlsWork.java:19]: a screwdriver click should be able to chance it

[the remark this answers is not in the document]
Comment (johnch18) [reply src/main/java/gregtech/common/covers/GT_Cover_ControlsWork.java:19]: Gotcha, I will work on that in a bit. I'm not familiar enough with the cover gui interface to do that but I'll take a look at it tonight.

[the remark this answers is not in the document]
Comment (johnch18) [reply src/main/java/gregtech/common/covers/GT_Cover_ControlsWork.java]: Okay, so I added two extra "safe" modes corresponding to normal and inverted behavior and I updated the GUI accordingly.

[the remark this answers is not in the document]
Comment (johnch18) [reply src/main/java/gregtech/common/covers/GT_Cover_ControlsWork.java]: I made sure that the safe modes use 3 and 4 for backwards compatibility. I wasn't sure how to setup the lang file stuff for this.

**Comment by johnch18** [comment]
> looks fine as it is now but
> having to renable save mode and the machine trough the cover is ennoying compared to a swing with the mallet
> and what would be realy nice to have is that the status of mWasShutdown to be displayed when using the tricorder so people can now if a lack of power disabled the machine

Gotcha, I'll get on it. I'll see about showing it in WAILA as well

**Comment by johnch18** [comment]
I'm just putting this here as a note for myself, I need to modify GT_Utility.getCoordinateScan

**Comment by Dream-Master** [comment]
@johnch18 still wip or is this pr done ?

**Comment by johnch18** [comment]
Still WIP, have college stuff to attend to

**Comment by johnch18** [comment]
I made it so that:
- the player receives a chat if a machine shuts down with its name and position.
- the scanner can see if an involuntary shutdown occurred. As well as if the device was disabled.

**Comment by Dream-Master** [comment]
@johnch18 let me know when it is done.

**Comment by johnch18** [comment]
It is as far as things that can be done on this end, barring any other modifications being suggested. Only thing I could really add is WAILA integration but that's in a completely different mod.

**Review by botn365 — APPROVED** [review]

**Pull request merged**